content/blog/gsoc-25-netjsongraph.rst

```diff
@@ -0,0 +1,225 @@
+GSoC 2025: Improving the netjsongraph.js Visualization Library
+==============================================================
+
+:date: 2025-09-22
+:author: cestercian
+:tags: gsoc, netjsongraph.js, visualization, new-features
+:category: gsoc
+:lang: en
+:image_url: https://openwisp.org/images/default-preview.webp
+
+.. image:: {static}/images/default-preview.webp
+    :alt: Google Summer of Code, Improving netjsongraph.js Visualization Library
+    :align: center
+
+Project Overview
+----------------
+
+This project focused on enhancing ``netjsongraph.js``, a powerful JavaScript library for visualizing network topologies. The primary goal was to improve the library's functionality, user experience (UX), and resiliency, making it more robust and capable for complex, real-world applications like those in OpenWISP. The work involved implementing several key features, from preventing visual clutter on maps to adding new data visualization capabilities and hardening the library against invalid data.
```

Comment:
Link the repo instead of flagging it as code please.

Reply:
Try to break down into a list to make it easy to read.

We worked on the following macro categories:

- Fix outstanding issues like the clustering
- Implement missing features
- Fix a number of live demos
- Improve the docs

I think this should be reflected in the docs.

Did I forget anything? If so, feel free to add.